index/secondary_index_manager.hh

```diff
@@ -116,6 +127,13 @@ public:
     bool is_index(view_ptr) const;
     bool is_index(const schema& s) const;
     bool is_global_index(const schema& s) const;
+    index_metrics* get_index_metrics(const sstring& index_name) {
```

Comment:
You had an option for pointer, or shared_ptr.
I'm not against pointers, but notice that you cannot store that pointer or use it after preemption so there will be no way someone will delete it.
Consider using shared_ptr